src/eigenlayer/ObolEigenLayerPodController.sol

```diff
@@ -4,173 +4,155 @@ pragma solidity 0.8.19;
+  error Invalid_FeeShare();
+  /// @dev user unauthorized
+  error Unauthorized();
+  /// @dev contract already initialized
+  error AlreadyInitialized();
+
+  /// @dev Emiited on intialize
+  event Initialized(address eigenPod, address owner);
+
+  /// -----------------------------------------------------------------------
+  /// libraries
+  /// -----------------------------------------------------------------------
+  using SafeTransferLib for address;
+  using SafeTransferLib for ERC20;
+
+  uint256 internal constant PERCENTAGE_SCALE = 1e5;
+
+  /// -----------------------------------------------------------------------
+  /// storage - immutables
+  /// -----------------------------------------------------------------------
+
+  /// @notice address of Eigenlayer delegation manager
+  /// @dev This is the address of the delegation manager transparent proxy
+  address public immutable eigenLayerDelegationManager;
+
+  /// @notice address of EigenLayerPod Manager
+  /// @dev this is the pod manager transparent proxy
+  IEigenPodManager public immutable eigenLayerPodManager;
+
+  /// @notice address of delay withdrawal router
+  IDelayedWithdrawalRouter public immutable delayedWithdrawalRouter;
+
+  /// @notice fee address
+  address public immutable feeRecipient;
+
+  /// @notice fee share. Represented as an integer from 1->10000 (100%)
+  uint256 public immutable feeShare;
+
+  /// -----------------------------------------------------------------------
+  /// storage
+  /// -----------------------------------------------------------------------
+
+  /// @notice address of deployed Eigen pod
+  address public eigenPod;
+
+  /// @notice address of a withdrawalAddress
+  address public withdrawalAddress;
+
+  /// @notice address of owner
+  address public owner;
+
+  modifier onlyOwner() {
+    if (msg.sender != owner) revert Unauthorized();
+    _;
+  }
+
+  constructor(
+    address recipient,
+    uint256 share,
+    address delegationManager,
+    address eigenPodManager,
+    address withdrawalRouter
+  ) {
+    if (recipient != address(0) && share == 0) revert Invalid_FeeSetup();
+    if (share > PERCENTAGE_SCALE) revert Invalid_FeeShare();
+
+    feeRecipient = recipient;
+    feeShare = share;
+    eigenLayerDelegationManager = delegationManager;
+    eigenLayerPodManager = IEigenPodManager(eigenPodManager);
+    delayedWithdrawalRouter = IDelayedWithdrawalRouter(withdrawalRouter);
+  }
+
+  /// @dev Enables contract to receive ETH
+  // defined on the clone implementation
+  // receive() external payable {}
+
+  /// @notice initializes the controller
+  /// @param _owner address of the controller owner
+  /// @param _withdrawalAddress address to receive funds
+  function initialize(address _owner, address _withdrawalAddress) external {
+    if (owner != address(0)) revert AlreadyInitialized();
+
+    eigenPod = eigenLayerPodManager.createPod();
+    owner = _owner;
+    withdrawalAddress = _withdrawalAddress;
```

Comment:
should we check for address(0) ?

Reply:
it's checked in the factory

Reply:
Could anyone do something weird like deploy a clone of this contract by using something other than our factory contract for doing so and manage to do something malicious? (In this case, I think subverting the factory to set your withdrawal address to 0 isn't smart/malicious, so not a problem, just asking here)

Reply:
Not possible, since the deployer address is included in `create2`